jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/ResourceService.java

```diff
@@ -14,241 +14,167 @@
-            _cacheControl = null;
-        if (cacheControl.getHeader() != HttpHeader.CACHE_CONTROL)
-            throw new IllegalArgumentException("!Cache-Control");
-        _cacheControl = cacheControl instanceof PreEncodedHttpField
-            ? cacheControl
-            : new PreEncodedHttpField(cacheControl.getHeader(), cacheControl.getValue());
+        return _cacheControl.getValue();
     }
 
+    /**
+     * @return file extensions that signify that a file is gzip compressed. Eg ".svgz"
+     */
     public List<String> getGzipEquivalentFileExtensions()
     {
         return _gzipEquivalentFileExtensions;
     }
 
-    public void setGzipEquivalentFileExtensions(List<String> gzipEquivalentFileExtensions)
+    /**
+     * @return Returns the stylesheet as a Resource.
+     */
+    public Resource getStylesheet()
     {
-        _gzipEquivalentFileExtensions = gzipEquivalentFileExtensions;
-    }
-
-    /* TODO
-    public boolean doGet(Request request, Response response) throws IOException
-    {
-        String servletPath = null;
-        String pathInfo = null;
-        Enumeration<String> reqRanges = null;
-        boolean included = request.getAttribute(RequestDispatcher.INCLUDE_REQUEST_URI) != null;
-        if (included)
+        if (_stylesheet != null)
         {
-            servletPath = _pathInfoOnly ? "/" : (String)request.getAttribute(RequestDispatcher.INCLUDE_SERVLET_PATH);
-            pathInfo = (String)request.getAttribute(RequestDispatcher.INCLUDE_PATH_INFO);
-            if (servletPath == null)
-            {
-                servletPath = request.getServletPath();
-                pathInfo = request.getPathInfo();
-            }
+            return _stylesheet;
         }
         else
         {
-            servletPath = _pathInfoOnly ? "/" : request.getServletPath();
-            pathInfo = request.getPathInfo();
-
-            // Is this a Range request?
-            reqRanges = request.getHeaders(HttpHeader.RANGE.asString());
-            if (!hasDefinedRange(reqRanges))
-                reqRanges = null;
+            if (_defaultStylesheet == null)
+            {
+                _defaultStylesheet = getDefaultStylesheet();
+            }
+            return _defaultStylesheet;
         }
+    }
 
-        String pathInContext = URIUtil.addPaths(servletPath, pathInfo);
+    public static Resource getDefaultStylesheet()
+    {
+        // TODO the returned path should point to the classpath.
+        // This points to a non-existent file '/jetty-dir.css'.
+        return Resource.newResource(Path.of("/jetty-dir.css"));
+    }
 
-        boolean endsWithSlash = (pathInfo == null ? (_pathInfoOnly ? "" : servletPath) : pathInfo).endsWith(URIUtil.SLASH);
-        boolean checkPrecompressedVariants = _precompressedFormats.length > 0 && !endsWithSlash && !included && reqRanges == null;
+    public void doGet(GenericRequest request, GenericResponse response, Callback callback, HttpContent content) throws Exception
```

Comment:
I don't t hink there is need of `GenericRequest` and `GenericResponse`.  Rather than always wrap both core and servlet objects to call this service, we should use core APIs and just wrap the servlet objects to look like core request/responses.

The reason I prefer this is that ultimately we will need those wrappers anyway to do such  things as cross context dispatch, so they may as well be used for calling this service and then not complicate this service when being used as core handler